### PR TITLE
MWPW-135444 Integrate Franklin's bulk preview and publish API

### PR DESCRIPTION
**Pull request by raga-adbe-gh**
Added a bulkPreviewPublish / bulkJobStatus that uses the franklin bulk apis.
Env- 
```
BULK_PREVIEW_CHECK_INTERVAL=8
MAX_BULK_PREVIEW_CHECKS=100
```
Tested with cc-pink - 
Preview - 150 files - https://admin.hlx.page/job/adobecom/cc-pink/main/preview/job-2023-10-07t17-00-33-769z
Publish - https://admin.hlx.page/job/adobecom/cc-pink/main/publish/job-2023-10-11t03-58-25-403z

### CODE REVIEW

**Review by arshadparwaiz — APPROVED** [review]

File: actions/promote/worker.js [hunk 1]
```diff
@@ -179,7 +179,7 @@ async function promoteFloodgatedFiles(doPublish, batchManager, appConfig) {
     let previewStatuses = [];
     let publishStatuses = [];
     if (ENABLE_HLX_PREVIEW) {
-        previewStatuses = await previewOrPublishPages(PREVIEW);
+        previewStatuses = await bulkPreviewOrPublish(PREVIEW);
```
Comment (sukamat) [inline actions/promote/worker.js]: There is another call happening for PUBLISH. I would assume, that should call the new API as well?
https://github.com/adobecom/milo-fg/blob/b458ed9b498ec3251c3e6ce647ecb5a4d9f2b13d/actions/promote/worker.js#L189

**Review by sukamat — COMMENTED** [review]
@raga-adbe-gh Looks good. One question about the PUBLISH call.

As discussed offline, we should test this with a larger set to understand the delay/wait that the API adds if a job goes beyond 15min. Given the batched approach with only 150 pages called per API call, I would assume we should be good on PROMOTE.

But the number of paths will be higher with bulk preview API call on COPY action ~500-1000 and those are not batched on our end. So we should definitely test that scenario.

File: actions/promote/worker.js (hunk 1 above)
Comment (raga-adbe-gh) [reply actions/promote/worker.js]: Thank you @sukamat . As suggested updated code with bulk publish api.

File: actions/helixUtils.js [hunk 2]
```diff
@@ -18,33 +18,123 @@
...
+            logger.info(`Status call response ${response.ok} with status ${response.status} `);
+            if (!response.ok && retryAttempt <= appConfig.getConfig().maxBulkPreviewChecks) {
+                await delay(appConfig.getConfig().bulkPreviewCheckInterval * 1000);
+                await this.bulkJobStatus(jobName, operation, repo, bulkPreviewStatus, retryAttempt + 1);
+            } else if (response.ok) {
+                const jobStatusJson = await response.json();
+                logger.info(`${operation} progress ${JSON.stringify(jobStatusJson.progress)}`);
+                jobStatusJson.data?.resources?.forEach((rs) => {
+                    if (operation === LIVE) {
+                        bulkPreviewStatus[rs.webPath] = { success: JOB_STATUS_CODES.includes(rs.status) };
```
Comment (sukamat) [inline actions/helixUtils.js]: @raga-adbe-gh The Publish Job Status API has been updated to use `path`. So we can remove this separate check.

**Review by sukamat — CHANGES_REQUESTED** [review]

File: actions/helixUtils.js (hunk 2 above)
Comment (raga-adbe-gh) [reply actions/helixUtils.js]: Thanks @sukamat  have updated the PR and also merged fgColor changes

**Review by sukamat — APPROVED** [review]

**Pull request merged**